docs/guides/typescript.md

```diff
@@ -399,6 +399,74 @@ A detailed explanation on the slices pattern can be found [here](./slices-patter
 
 If you have some middlewares then replace `StateCreator<MyState, [], [], MySlice>` with `StateCreator<MyState, Mutators, [], MySlice>`. For example, if you are using `devtools` then it will be `StateCreator<MyState, [["zustand/devtools", never]], [], MySlice>`. See the ["Middlewares and their mutators reference"](#middlewares-and-their-mutators-reference) section for a list of all mutators.
 
+### Using a vanilla store as a bound store
+
+Create your vanilla store:
+
+```ts
+import { createStore } from 'zustand/vanilla'
+
+interface BearState {
+  bears: number
+  increase: (by: number) => void
+}
+
+export const initialBearState = { bears: 0 }
+export const vanillaBearStore = createStore<BearState>((set, getState) => ({
+  ...initialBearState,
+  increase: (by) => set((state) => ({ bears: state.bears + by })),
+}))
+```
+
+Create a hook to provide a bound store to be used in your component:
+
+```ts
+import { useStore } from 'zustand'
+
+export function useBoundBearStore(): BearState
+export function useBoundBearStore<T>(
+  selector: (state: BearState) => T,
+  equals?: (a: T, b: T) => boolean
+): T
+export function useBoundBearStore(selector?: any, equals?: any) {
+  return useStore(vanillaBearStore, selector, equals)
+}
+```
+
+> **_NOTE:_** We prefer function overloading here, as this closely follows the definition of `useStore` itself.  
+> If you are not familiar with this pattern, just have a look here: [Typescript Docs](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads)
+
+Now you can access your vanilla store (e.g. in your tests) like:
+
+```ts
+import { vanillaBearStore, initialBearState } from './BearStore'
```

Comment:
Sorry for being so late, overall a great PR and discussion,
however here you import `initialBearState`, but you use `initialState`.

It's not a critical change, but it can be updated :)